### PR TITLE
event filtering implementation

### PR DESCRIPTION
**Pull request by jumpingfool**
Fixed issue #1 and added an implementation of simple event filtering as per the specification here (https://docs.microsoft.com/en-us/azure/event-grid/event-filtering)

includedEventTypes, subjectBeginsWith, subjectEndsWith and isSubjectCaseSensitive are supported however I have not yet got around to implementing advancedFilters

### CODE REVIEW

**Comment by jumpingfool** [comment]
Added support for advanced filtering of eventGridEvents

**Comment by pmcilreavy** [comment]
Would you mind updating the ReadMe with a some usage examples.

File: src/AzureEventGridSimulator/appsettings.json [hunk 1]
```diff
@@ -7,7 +7,27 @@
       "subscribers": [
         {
           "name": "LocalAzureFunctionSubscription",
-          "endpoint": "http://localhost:7071/runtime/webhooks/EventGrid?functionName=PersistEventToDb"
+          "endpoint": "http://localhost:7071/runtime/webhooks/EventGrid?functionName=PersistEventToDb",
+          "filter": {
```
Comment (pmcilreavy) [inline src/AzureEventGridSimulator/appsettings.json]: I think for the F5 experience, I don't think we should have any pre-configured filters in here. Can we remove `filter` from here and add some examples in the `ReadMe` file.

**Comment by jumpingfool** [comment]
Hopefully these changes will be acceptable. Many thanks for getting around to looking at this PR, I know how busy life can get sometimes. Let me know if there is anything more I can do.

**Pull request merged**